Taken from LFS: http://www.linuxfromscratch.org/patches/lfs/cvs/unstable/

Submitted By: Jeremy Utley <jeremy@linuxfromscratch.org>
Initial Package Version: 2.15.94.0.1
Upstream Status: From Upstream
Origin: ML Post by HJ Lu - http://sources.redhat.com/ml/binutils/2004-11/msg00332.html
Description: With this version of binutils, TLS information is lost when
stripping libc.a.  This patch fixes the problem.  Binutils bugzilla id 574.

diff -Naur binutils-2.15.94.0.1/bfd/elfcode.h binutils-2.15.94.0.1-new/bfd/elfcode.h
--- binutils-2.15.94.0.1/bfd/elfcode.h	2004-11-22 20:33:31.000000000 +0000
+++ binutils-2.15.94.0.1-new/bfd/elfcode.h	2004-12-04 20:27:06.800747251 +0000
@@ -1158,6 +1158,9 @@
 	    case STT_OBJECT:
 	      sym->symbol.flags |= BSF_OBJECT;
 	      break;
+	    case STT_TLS:
+	      sym->symbol.flags |= BSF_THREAD_LOCAL;
+	      break;
 	    }
 
 	  if (dynamic)
